### PR TITLE
Coconut Proof allows empty blinding messages

### PR DESCRIPTION
**Pull request by 0xvon**
## Issue

https://github.com/docknetwork/crypto/issues/20

## What I do

- change UnpackedBlindedMessages impl to return empty proof even if blinding message is empty
- change ps_proof's proving to use the index not i but j

### CODE REVIEW

File: coconut/src/proof/messages_pok/mod.rs [hunk 1]
```diff
@@ -558,14 +558,23 @@ mod tests {
...
-        assert_eq!(
-            MessagesPoKGenerator::init(&mut rng, &[], &params, &h),
-            Err(MessagesPoKError::MessageInputError(
-                MessageUnpackingError::NoMessagesProvided
-            ))
-        );
+        let pok = MessagesPoKGenerator::init(&mut rng, &[], &params, &h)
+            .unwrap();
+
+        let mut chal_bytes_prover = vec![];
```
Comment (lovesh) [inline coconut/src/proof/messages_pok/mod.rs:567]: As you are using the same variables for prover's and verifier's challenge, they should not have suffix `_prover`, `chal_bytes` is good.

File: coconut/src/proof/mod.rs [hunk 2]
```diff
@@ -178,7 +178,7 @@ impl<'pair, Pair, F: PrimeField> UnpackedBlindedMessages<'pair, Pair, F> {
         let (paired, (msgs, blindings)): (Vec<_>, _) =
             process_results(paired, |iter| iter.unzip())?;
         if paired.is_empty() {
```
Comment (lovesh) [inline coconut/src/proof/mod.rs:180]: You can remove this `if` block as the last `else` block will return the desired result.

File: coconut/src/proof/signature_pok/mod.rs [hunk 3]
```diff
@@ -347,12 +347,20 @@ mod tests {
...
 
-        assert_eq!(
-            SignaturePoKGenerator::init(&mut rng, &[], &sig, &pk, &params),
-            Err(SignaturePoKError::MessageInputError(
-                MessageUnpackingError::NoMessagesProvided
-            ))
-        );
+        let pok = SignaturePoKGenerator::init(&mut rng, &[], &sig, &pk, &params).unwrap();
+
+        let mut chal_bytes_prover = vec![];
```
Comment (lovesh) [inline coconut/src/proof/signature_pok/mod.rs:352]: Similar comment as above on naming.

**Review by lovesh — CHANGES_REQUESTED** [review]

**Comment by 0xvon** [comment]
@lovesh Thank you so much for your quick feedbacks! I modified for all of your comments!

**Comment by lovesh** [comment]
@0xvon I have fixed the failing tests in [this commit](https://github.com/docknetwork/crypto/commit/05f02df4d773c7e59233faeed74c758904180824) but I am not able to edit this PR even when this PR is marked appropriately. Can you please get these changes in your PR? Thanks

**Comment by 0xvon** [comment]
> @0xvon I have fixed the failing tests in [this commit](https://github.com/docknetwork/crypto/commit/05f02df4d773c7e59233faeed74c758904180824) but I am not able to edit this PR even when this PR is marked appropriately. Can you please get these changes in your PR? Thanks

I pushed!

**Pull request merged**